<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Richel J.C. Bilderbeek" />

<meta name="date" content="2017-11-22" />

<title>beautier demo</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">beautier demo</h1>
<h4 class="author"><em>Richel J.C. Bilderbeek</em></h4>
<h4 class="date"><em>2017-11-22</em></h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The purpose of <code>beautier</code> is to create a valid <code>BEAST2</code> XML input file from its function argument. In this way, a scientific pipeline using <code>BEAST2</code> can be fully scripted, instead of using <code>BEAUti</code>â€™s GUI.</p>
</div>
<div id="demonstration" class="section level1">
<h1>Demonstration</h1>
<p>First, <code>beautier</code> is loaded:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(beautier)</code></pre></div>
<p>A BEAST2 XML input file needs an alignment (as BEAST2 infers phylogenies and parameters on DNA sequences). in this demonstration, a FASTA file with a random alignment is created:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fasta_filename &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="dt">pattern =</span> <span class="st">&quot;demo&quot;</span>, <span class="dt">fileext =</span> <span class="st">&quot;.fas&quot;</span>)

beautier<span class="op">::</span><span class="kw">create_random_fasta</span>(
  <span class="dt">n_taxa =</span> <span class="dv">5</span>, <span class="co"># species</span>
  <span class="dt">sequence_length =</span> <span class="dv">20</span>, <span class="co"># DNA nucleotides</span>
  <span class="dt">filename =</span> fasta_filename
)</code></pre></div>
<p>We can display the alignment in the file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">image</span>(ape<span class="op">::</span><span class="kw">read.FASTA</span>(fasta_filename))</code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC8VBMVEUAAAAAAAoAABoAAB0AADEAADMAAE4AAIgAAJEAAP8AJwAAMwAAaQAAhwAAiAAAxAAA/QAA/wABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJAAAJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXAAAXFxcYGBgZGRkaGhobGxscHBwdHR0eAAAfHx8gICAhISEiIiIkJCQlJSUnJycoKAAoKCgpKSkqKiorAAArKyssLCwtLS0uLi4vLy8wMDAxMQAxMTEyMjIzAAAzMwAzMzM0NDQ1NTU2NjY3Nzc4ODg7Ozs8PDw+Pj4/Pz9AQEBBQUFDQ0NGRkZHR0dISEhJSUlMTExNTU1PT09QUFBRAABRUVFSUlJTU1NUVFRVVVVWVlZYWFhZWVlaWlpcXFxdXV1eXl5gYGBhYWFiYmJkZGRlZWVmZmZoaGhra2tsbABsbGxtbW1ubm5wcHBxcXFycnJ0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICCgoKDg4OEhACEhISFhYWGhoaHh4eIAACIiACIiIiJiYmKioqLi4uMjIyNjY2Pj4+QkJCTk5OUlJSVlZWWlpaXAACYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKCioqKjo6OkpKSmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrK1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2/v7/AwMDBwcHCwsLDw8PExMTGxsbHx8fIyMjKygDKysrMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLU1NTV1dXX19fY2NjZ2dna2trb29vc3Nze3t7f39/g4ODh4eHi4uLj4+Pk5OTo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39wD39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7/AAD//wD////ZvMJoAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAIV0lEQVR4nO3deVwUZRzH8ac7uw3NDIHUDjMpS7PSwi4rNVGxw+i0w0zJK1Mzr7S0skMtu7PD2zJNLa0kqDCLyDIrM8+8DzRLZXP+amZ259l57T7PfJ946cLW9/PH7jL8fGDeAgu7AyMsFpio6HegskcgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABwJoY7GvxQGDq/2DvwQM/uwfXKOfC33nH9xU3h3wdyCAep93TbRTftIPZl98reyq9IAVaxwd7aj2+rkfDjos2iH9y70Hvg4EUK/X/4526RLfa3LEcP9g66/3yf6oGd2+MPeMWpc9sj66IeWEqrLj20a3z8ipU6fN1JB8ueTQKtGO7BvZepeI9FQ5diahQGtS0i/0D2qAxlVvM3LSgLqZq+UWDdBD4taXx3cSw+QGNdBXM2fObNjWvlhajp1JKNC46q+IEt+gGqhIDCyzr1Y06CgH1UCF4kXnavCJ8suXGsip+b3l3JmEArXOLa01yDeoBsq+vMy9fidnu7dJDZRzpXu1vtkEb0uSA/0qpludGvkGlUDbxctxKyqB9lQfGzuX5EBPpO+w3hWLooNKoG/FAvvyxzl28g5dCbRMzIx9y0kOlNXVskpTfXe+SqAvRYF9Odi513nXG1QCfS8+jH3LyQ30Q/i+9pzovbISaIN4M3zjz2rBQKXipfCN6dO8ueQGGpL2WWFh4SjxhRxUf5G+oF34ugh8BFnNwt8y7jm1t7cluYEa3+NcbjtZ7o4G6H0xxh1siYDeC3+svSA+9bYkNdAiMcO9zq231xvUfKPYW3R4ZurjmT0bAKBQd9Hp1YndTsyT/zJZgG5/NtpZHlC/1FL3eoLI9wZbP/ic7Enfjxpzc8+pd/N71ohCb0PKscfJjvH9qDEtN7POdW+VyZdLDj4i2uGVFmjBIF9Dt+sHJ/kHRwesOMo/OEU/t2WIf7Cg3Hvgi48HgQgEIhCIQCACgQgEIhCIQCACgQgEIhCIQCACgQgEIhBoPwM93DJJare2YoCyvQdG5QOf3kOq8iHTfbq8B2njNsi1/o6takz6txr31r8nEIEIRCACEciyutzmXu1p24VAQUADBIGUZQshtljW3LrnE0jZxk4d14as3+rOziaQOudT7K9W/S0CBQD1brObQAFAs05baREoAKhf5Ej2OQRSA/2cb9fshvx1BFKUl1XsHjLGTzFNBY0zthLINAIRiEAEIlBlBsIesQ76ySqxaYG0HvKfxr2GQAQiEIEIRCAr8ojiqKa1Lhm7m0A6oME1n/54xMl9CaTIeVZjU+pQ+9bIk3YRKD7nWY1VTYvsW1PESgIpijxxaO1s36SMQHqgklYZC/k1SAu0MS/ljhUWgXRAv2Q2/8Z9gUBKoFBWx10WgTTlZRV/Lvq/7bSTQIoKGmc8H3nicC2BUAQiEIEIRKDKDBT3jmiBtLu/L3Z39wdQ3ASBCEQgAhGIQJZ8wKzfAAIFAIXmpxFInfu7GrMzhCCQOvd3NTYvWdKQQJoiX4OaEIhABCIQgSohUOR3NQikK/K7GgQyikAEIhCBCESgJAaK8/CK20stkN7We0UclLY4KfkaAhGIQAQiEIGs8MMdW/s0SmsxnUBaoLvrv/lxd/ERgRQ5T/ssF5PsW827EEiR87TP8m7OUfYduhJIVfgh19+LRmYUEEgPNDr1zK47CKQHsqxQ51sJpAGad79z421RRiA10AL3DM69MvkRpCovq3jXFU0mfzKw2isEUuU87bP6vsy0qydbBIIRiEAEIhCB/i9AsTtXJQ4odnECEYhABCIQgf5dzuNBs9y/K9CdQDqgMfXfsCskkCL3l1l63Oi9RKDY3F9mub7PppLNBFLnfIo1bFpDiDvXE0gD9GfqTUu3zT19fz9o/98BCveS2EKgIKD5YhmB1ED5NfPtG8Nr7+enff4jQHlZxbtbnjt2/qM1XuPXIFXOsxpr8xqkXzfDIhCMQAQiEIEIVImB3uppUu3zmxl2Rj3TyYbpppMXpdjvQt+tFQNkVrsvTSeHjjKdnHez6WRpmumkE4FABAIRCEQgEIFABAIRCFQhQB2+Mp18bKzp5IJbTCd31TWddKoQoI0h08kdO00nyzYZv/kNxpNWBQElUwQCEQhEIBCBQAQCEQhEIBCBQAQCEQhEIBCBQIkHksdUo8KnEbAmtqydU2QyCReWJ1w2WdMr8UDymOrgvNMITBUPf3BH6mKDSbiwPOEyXjNa4oHkMdWBydMIXN7ZsvY27WEwiRYukydchmv6SjyQPKY6MO80AqvELPtycAM8CRde7Z1wGa/pK/FA8phqlPsHmBeJYvvyjZS9cNJsYeeEy0ZreiUcKHpMNcrd7XlimX05XayDk0YLuydcNlrTq4Lu5p1jqlGRjyDnz4KMT9kNJw0Wjpxw2WhNrwoCmu/+Jwbn7vYqMde+HHY2nsQLeydcNlrTK+FA0WOqUeHdbm5/YxO64gE8CReOnnDZZE2vhAOVyWOqUWGgKSnjFvZMDT4ezJ2ECxfJEy6brOmV+E8xeUw1KvKJM6nFqe0XmkyihcfLEy6brOnFn8VABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCEQgEIFABAIRCPQPkdCsw1CD6IMAAAAASUVORK5CYII=" /><!-- --></p>
<p>Now we can create our XML file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># The name of the file you intend to let BEAST2 run</span>
output_xml_filename &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="dt">pattern =</span> <span class="st">&quot;demo&quot;</span>, <span class="dt">fileext =</span> <span class="st">&quot;.xml&quot;</span>)

beautier<span class="op">::</span><span class="kw">create_beast2_input_file</span>(
  fasta_filename,
  output_xml_filename
)</code></pre></div>
<p>(note: the line <code>tree_priors = create_yule_tree_prior(birth_rate_distribution = create_uniform_distr(id = 1))</code> will be removed somewhere close in the future)</p>
<p>The file indeed is a BEAST2 input file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">cat</span>(<span class="kw">readLines</span>(output_xml_filename), <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">'</span><span class="ch">\n</span><span class="st">'</span>)</code></pre></div>
<pre><code>## &lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot; standalone=&quot;no&quot;?&gt;&lt;beast beautitemplate='Standard' beautistatus='' namespace=&quot;beast.core:beast.evolution.alignment:beast.evolution.tree.coalescent:beast.core.util:beast.evolution.nuc:beast.evolution.operators:beast.evolution.sitemodel:beast.evolution.substitutionmodel:beast.evolution.likelihood&quot; required=&quot;&quot; version=&quot;2.4&quot;&gt;
## 
## 
##     &lt;data
## id=&quot;demo12ed256cbd28&quot;
## name=&quot;alignment&quot;&gt;
##                     &lt;sequence id=&quot;seq_t1&quot; taxon=&quot;t1&quot; totalcount=&quot;4&quot; value=&quot;ccggattagctaacgcctgc&quot;/&gt;
##                     &lt;sequence id=&quot;seq_t2&quot; taxon=&quot;t2&quot; totalcount=&quot;4&quot; value=&quot;gcggagcggttaacgcctcc&quot;/&gt;
##                     &lt;sequence id=&quot;seq_t3&quot; taxon=&quot;t3&quot; totalcount=&quot;4&quot; value=&quot;gggtattacggcaaaaacta&quot;/&gt;
##                     &lt;sequence id=&quot;seq_t4&quot; taxon=&quot;t4&quot; totalcount=&quot;4&quot; value=&quot;ctgcgggagataacctgccg&quot;/&gt;
##                     &lt;sequence id=&quot;seq_t5&quot; taxon=&quot;t5&quot; totalcount=&quot;4&quot; value=&quot;gggtattacctcagtaacta&quot;/&gt;
##                 &lt;/data&gt;
## 
## 
##     
## 
## 
##     
## 
## 
##     
## &lt;map name=&quot;Uniform&quot; &gt;beast.math.distributions.Uniform&lt;/map&gt;
## &lt;map name=&quot;Exponential&quot; &gt;beast.math.distributions.Exponential&lt;/map&gt;
## &lt;map name=&quot;LogNormal&quot; &gt;beast.math.distributions.LogNormalDistributionModel&lt;/map&gt;
## &lt;map name=&quot;Normal&quot; &gt;beast.math.distributions.Normal&lt;/map&gt;
## &lt;map name=&quot;Beta&quot; &gt;beast.math.distributions.Beta&lt;/map&gt;
## &lt;map name=&quot;Gamma&quot; &gt;beast.math.distributions.Gamma&lt;/map&gt;
## &lt;map name=&quot;LaplaceDistribution&quot; &gt;beast.math.distributions.LaplaceDistribution&lt;/map&gt;
## &lt;map name=&quot;prior&quot; &gt;beast.math.distributions.Prior&lt;/map&gt;
## &lt;map name=&quot;InverseGamma&quot; &gt;beast.math.distributions.InverseGamma&lt;/map&gt;
## &lt;map name=&quot;OneOnX&quot; &gt;beast.math.distributions.OneOnX&lt;/map&gt;
## 
## 
## &lt;run id=&quot;mcmc&quot; spec=&quot;MCMC&quot; chainLength=&quot;10000000&quot;&gt;
##     &lt;state id=&quot;state&quot; storeEvery=&quot;5000&quot;&gt;
##         &lt;tree id=&quot;Tree.t:demo12ed256cbd28&quot; name=&quot;stateNode&quot;&gt;
##             &lt;taxonset id=&quot;TaxonSet.demo12ed256cbd28&quot; spec=&quot;TaxonSet&quot;&gt;
##                 &lt;alignment idref=&quot;demo12ed256cbd28&quot;/&gt;
##             &lt;/taxonset&gt;
##         &lt;/tree&gt;
##         &lt;parameter id=&quot;birthRate.t:demo12ed256cbd28&quot; name=&quot;stateNode&quot;&gt;1.0&lt;/parameter&gt;
##     &lt;/state&gt;
## 
##     &lt;init id=&quot;RandomTree.t:demo12ed256cbd28&quot; spec=&quot;beast.evolution.tree.RandomTree&quot; estimate=&quot;false&quot; initial=&quot;@Tree.t:demo12ed256cbd28&quot; taxa=&quot;@demo12ed256cbd28&quot;&gt;
##         &lt;populationModel id=&quot;ConstantPopulation0.t:demo12ed256cbd28&quot; spec=&quot;ConstantPopulation&quot;&gt;
##             &lt;parameter id=&quot;randomPopSize.t:demo12ed256cbd28&quot; name=&quot;popSize&quot;&gt;1.0&lt;/parameter&gt;
##         &lt;/populationModel&gt;
##     &lt;/init&gt;
## 
##     &lt;distribution id=&quot;posterior&quot; spec=&quot;util.CompoundDistribution&quot;&gt;
##         &lt;distribution id=&quot;prior&quot; spec=&quot;util.CompoundDistribution&quot;&gt;
##             &lt;distribution id=&quot;YuleModel.t:demo12ed256cbd28&quot; spec=&quot;beast.evolution.speciation.YuleModel&quot; birthDiffRate=&quot;@birthRate.t:demo12ed256cbd28&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot;/&gt;
##             &lt;prior id=&quot;YuleBirthRatePrior.t:demo12ed256cbd28&quot; name=&quot;distribution&quot; x=&quot;@birthRate.t:demo12ed256cbd28&quot;&gt;
##                 &lt;Uniform id=&quot;Uniform.100&quot; name=&quot;distr&quot; upper=&quot;Infinity&quot;/&gt;
##             &lt;/prior&gt;
##         &lt;/distribution&gt;
##         &lt;distribution id=&quot;likelihood&quot; spec=&quot;util.CompoundDistribution&quot; useThreads=&quot;true&quot;&gt;
##             &lt;distribution id=&quot;treeLikelihood.demo12ed256cbd28&quot; spec=&quot;ThreadedTreeLikelihood&quot; data=&quot;@demo12ed256cbd28&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot;&gt;
##                 &lt;siteModel id=&quot;SiteModel.s:demo12ed256cbd28&quot; spec=&quot;SiteModel&quot;&gt;
##                     &lt;parameter id=&quot;mutationRate.s:demo12ed256cbd28&quot; estimate=&quot;false&quot; name=&quot;mutationRate&quot;&gt;1.0&lt;/parameter&gt;
##                     &lt;parameter id=&quot;gammaShape.s:demo12ed256cbd28&quot; estimate=&quot;false&quot; name=&quot;shape&quot;&gt;1.0&lt;/parameter&gt;
##                     &lt;parameter id=&quot;proportionInvariant.s:demo12ed256cbd28&quot; estimate=&quot;false&quot; lower=&quot;0.0&quot; name=&quot;proportionInvariant&quot; upper=&quot;1.0&quot;&gt;0.0&lt;/parameter&gt;
##                     &lt;substModel id=&quot;JC69.s:demo12ed256cbd28&quot; spec=&quot;JukesCantor&quot;/&gt;
##                 &lt;/siteModel&gt;
##                 &lt;branchRateModel id=&quot;StrictClock.c:demo12ed256cbd28&quot; spec=&quot;beast.evolution.branchratemodel.StrictClockModel&quot;&gt;
##                     &lt;parameter id=&quot;clockRate.c:demo12ed256cbd28&quot; estimate=&quot;false&quot; name=&quot;clock.rate&quot;&gt;1.0&lt;/parameter&gt;
##                 &lt;/branchRateModel&gt;
##             &lt;/distribution&gt;
##         &lt;/distribution&gt;
##     &lt;/distribution&gt;
## 
##     &lt;operator id=&quot;YuleBirthRateScaler.t:demo12ed256cbd28&quot; spec=&quot;ScaleOperator&quot; parameter=&quot;@birthRate.t:demo12ed256cbd28&quot; scaleFactor=&quot;0.75&quot; weight=&quot;3.0&quot;/&gt;
## 
##     &lt;operator id=&quot;YuleModelTreeScaler.t:demo12ed256cbd28&quot; spec=&quot;ScaleOperator&quot; scaleFactor=&quot;0.5&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot; weight=&quot;3.0&quot;/&gt;
## 
##     &lt;operator id=&quot;YuleModelTreeRootScaler.t:demo12ed256cbd28&quot; spec=&quot;ScaleOperator&quot; rootOnly=&quot;true&quot; scaleFactor=&quot;0.5&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot; weight=&quot;3.0&quot;/&gt;
## 
##     &lt;operator id=&quot;YuleModelUniformOperator.t:demo12ed256cbd28&quot; spec=&quot;Uniform&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot; weight=&quot;30.0&quot;/&gt;
## 
##     &lt;operator id=&quot;YuleModelSubtreeSlide.t:demo12ed256cbd28&quot; spec=&quot;SubtreeSlide&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot; weight=&quot;15.0&quot;/&gt;
## 
##     &lt;operator id=&quot;YuleModelNarrow.t:demo12ed256cbd28&quot; spec=&quot;Exchange&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot; weight=&quot;15.0&quot;/&gt;
## 
##     &lt;operator id=&quot;YuleModelWide.t:demo12ed256cbd28&quot; spec=&quot;Exchange&quot; isNarrow=&quot;false&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot; weight=&quot;3.0&quot;/&gt;
## 
##     &lt;operator id=&quot;YuleModelWilsonBalding.t:demo12ed256cbd28&quot; spec=&quot;WilsonBalding&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot; weight=&quot;3.0&quot;/&gt;
## 
##     &lt;logger id=&quot;tracelog&quot; fileName=&quot;demo12ed256cbd28.log&quot; logEvery=&quot;1000&quot; model=&quot;@posterior&quot; sanitiseHeaders=&quot;true&quot; sort=&quot;smart&quot;&gt;
##         &lt;log idref=&quot;posterior&quot;/&gt;
##         &lt;log idref=&quot;likelihood&quot;/&gt;
##         &lt;log idref=&quot;prior&quot;/&gt;
##         &lt;log idref=&quot;treeLikelihood.demo12ed256cbd28&quot;/&gt;
##         &lt;log id=&quot;TreeHeight.t:demo12ed256cbd28&quot; spec=&quot;beast.evolution.tree.TreeHeightLogger&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot;/&gt;
##         &lt;log idref=&quot;YuleModel.t:demo12ed256cbd28&quot;/&gt;
##         &lt;log idref=&quot;birthRate.t:demo12ed256cbd28&quot;/&gt;
##     &lt;/logger&gt;
## 
##     &lt;logger id=&quot;screenlog&quot; logEvery=&quot;1000&quot;&gt;
##         &lt;log idref=&quot;posterior&quot;/&gt;
##         &lt;log id=&quot;ESS.0&quot; spec=&quot;util.ESS&quot; arg=&quot;@posterior&quot;/&gt;
##         &lt;log idref=&quot;likelihood&quot;/&gt;
##         &lt;log idref=&quot;prior&quot;/&gt;
##     &lt;/logger&gt;
## 
##     &lt;logger id=&quot;treelog.t:demo12ed256cbd28&quot; fileName=&quot;$(tree).trees&quot; logEvery=&quot;1000&quot; mode=&quot;tree&quot;&gt;
##         &lt;log id=&quot;TreeWithMetaDataLogger.t:demo12ed256cbd28&quot; spec=&quot;beast.evolution.tree.TreeWithMetaDataLogger&quot; tree=&quot;@Tree.t:demo12ed256cbd28&quot;/&gt;
##     &lt;/logger&gt;
## 
## &lt;/run&gt;
## 
## &lt;/beast&gt;
## FALSE</code></pre>
<p>This XML input file can be read by BEAST2.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
