---
title: "beastscriptr demo"
author: "Richel J.C. Bilderbeek"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{beastscriptr demo}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# Introduction

The purpose of `beastscriptr` is to create a valid `BEAST2` XML input file
from its function argument. In this way, a scientific pipeline using 
`BEAST2` can be fully scripted, instead of using `BEAUti`'s GUI.

# Limitations

`BEAUti2` contains multiple tree priors. This package currently
only supports:

 * These tree priors:
    * the constant-rate birth-death model
    * the contant-population coalescent model
 * DNA data

# Demonstration

A BEAST2 XML input file needs an alignment (as BEAST2 infers
phylogenies and parameters on DNA sequences). in this demonstration,
a FASTA file with a random alignment is created:

```{r}
fasta_filename <- tempfile(pattern = "demo", fileext = ".fas")

beastscriptr::create_random_fasta(
  n_taxa = 5, # species
  sequence_length = 20, # DNA nucleotides
  filename = fasta_filename
)
```

We can display the alignment in the file:


```{r}
image(ape::read.FASTA(fasta_filename))
```

Now we can create our XML file:

```{r}
# The name of the file you intend to let BEAST2 run
output_xml_filename <- tempfile(pattern = "demo", fileext = ".xml")

beastscriptr::create_beast2_input_file(
  fasta_filename,
  output_xml_filename
)
```

The file indeed is a BEAST2 input file:

```{r}
print(readLines(output_xml_filename))
```

This XML input file can be read by BEAST2.
