---
title: "beastscriptr demo"
author: "Richel Bilderbeek"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{beastscriptr demo}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# Introduction

The purpose of `beastscriptr` is to create a valid `BEAST2` XML input file
from its function argument. In this way, a scientific pipeline using 
`BEAST2` can be fully scripted, instead of using `BEAUti2` its GUI.

# Limitations

`BEAUti2` contains multiple tree priors. This package currently
only supports:

 * These tree priors:
    * the constant-rate birth-death model
    * the contant-population coalescent model
 * DNA data

# Demonstration

The library is added first:

```{r}
library(beastscriptr)
```

A BEAST2 XML input file needs an alignment (as BEAST2 infers
phylogenies and parameters on DNA sequences). in this demonstration,
a random alignment is created:

```{r}
n_taxa <- 5
dna_sequence_length_nucleotides <- 20
mutation_rate <- 0.1

alignment <- create_random_alignment(
  n_taxa = n_taxa,
  sequence_length = dna_sequence_length_nucleotides,
  rate = mutation_rate
)

image(alignment)
```

This random alignment must be saved to a FASTA file:

```{r}
fasta_filename <- "demo.fas"

phangorn::write.phyDat(
  x = alignment, 
  file = fasta_filename, 
  format = "fasta"
)
if (file.exists(fasta_filename)) { 
  print("File created") 
} else {
  print("Error: file not created") 
}
```

We also need to define some parameters:

```{r}
# The MCMC chain length BEAST2 will have to use
mcmc_chainlength <- 10000000

# The tree prior BEAST2 will have to use, can be 'birth_death' or 'coalescent_constant_population'
tree_prior <- "birth_death"
# tree_prior <- "coalescent_constant_population"

# The date in YYYYMMDD format. 
# This is important, as BEAST2 and BEAUti2 can change formats
date_str <- "20151022"

# The name of the file you intend to let BEAST2 run
output_xml_filename <- "demo.xml"
```

Now we can create our XML file:

```{r}

beast_scriptr(
  input_fasta_filename = fasta_filename,
  mcmc_chainlength = mcmc_chainlength,
  tree_prior = tree_prior,
  date_str = date_str,
  output_xml_filename = output_xml_filename
)

if (file.exists(output_xml_filename)) { 
  print("File created") 
} else {
  print("Error: file not created") 
}
```

The file indeed is a BEAST2 input file:

```{r}
print(readLines(output_xml_filename))
```

This XML input file should be readable by BEAST2.

All left to do is cleaning up the files created by this demonstration.

```{r}
file.remove(fasta_filename)
file.remove(output_xml_filename)
```
