<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head><title>R: 'BEAUti' from R</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>
<h1> 'BEAUti' from R
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;beautier&rsquo; version 1.13</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
</ul>

<h2>Help Pages</h2>


<p style="text-align: center;">
<a href="#A">A</a>
<a href="#B">B</a>
<a href="#C">C</a>
<a href="#D">D</a>
<a href="#E">E</a>
<a href="#F">F</a>
<a href="#G">G</a>
<a href="#H">H</a>
<a href="#I">I</a>
<a href="#M">M</a>
<a href="#P">P</a>
<a href="#R">R</a>
<a href="#S">S</a>
<a href="#T">T</a>
<a href="#Y">Y</a>
</p>


<h2><a name="A">-- A --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="are_clock_models.html">are_clock_models</a></td>
<td>Determine if x consists out of clock_models objects</td></tr>
<tr><td style="width: 25%;"><a href="are_equal_xml_files.html">are_equal_xml_files</a></td>
<td>Determine if XML files result in equal trees</td></tr>
<tr><td style="width: 25%;"><a href="are_equal_xml_lines.html">are_equal_xml_lines</a></td>
<td>Determine if XML lines result in equal trees</td></tr>
<tr><td style="width: 25%;"><a href="are_equivalent_xml_files.html">are_equivalent_xml_files</a></td>
<td>Determine if XML files result in equivalent trees</td></tr>
<tr><td style="width: 25%;"><a href="are_equivalent_xml_lines.html">are_equivalent_xml_lines</a></td>
<td>Determine if XML lines result in equivalent trees</td></tr>
<tr><td style="width: 25%;"><a href="are_equivalent_xml_lines_all.html">are_equivalent_xml_lines_all</a></td>
<td>Determine if XML lines result in equivalent trees</td></tr>
<tr><td style="width: 25%;"><a href="are_equivalent_xml_lines_loggers.html">are_equivalent_xml_lines_loggers</a></td>
<td>Determine if XML operator lines result in equivalent trees</td></tr>
<tr><td style="width: 25%;"><a href="are_equivalent_xml_lines_operators.html">are_equivalent_xml_lines_operators</a></td>
<td>Determine if XML operator lines result in equivalent trees</td></tr>
<tr><td style="width: 25%;"><a href="are_equivalent_xml_lines_section.html">are_equivalent_xml_lines_section</a></td>
<td>Determine if XML lines result in equivalent trees</td></tr>
<tr><td style="width: 25%;"><a href="are_ids.html">are_ids</a></td>
<td>Determine if x consists out of IDs</td></tr>
<tr><td style="width: 25%;"><a href="are_init_clock_models.html">are_init_clock_models</a></td>
<td>Determine if x consists out of initialized clock_models objects</td></tr>
<tr><td style="width: 25%;"><a href="are_init_mrca_priors.html">are_init_mrca_priors</a></td>
<td>Determine if x consists out of initialized mrca_priors objects</td></tr>
<tr><td style="width: 25%;"><a href="are_init_site_models.html">are_init_site_models</a></td>
<td>Determine if x consists out of initialized site_models objects</td></tr>
<tr><td style="width: 25%;"><a href="are_init_tree_priors.html">are_init_tree_priors</a></td>
<td>Determine if x consists out of initialized tree_priors objects</td></tr>
<tr><td style="width: 25%;"><a href="are_mrca_align_ids_in_fastas.html">are_mrca_align_ids_in_fastas</a></td>
<td>Determine if the MRCA priors' alignment IDs are present in the FASTA files</td></tr>
<tr><td style="width: 25%;"><a href="are_mrca_priors.html">are_mrca_priors</a></td>
<td>Determine if x consists out of mrca_priors objects</td></tr>
<tr><td style="width: 25%;"><a href="are_mrca_taxa_names_in_fastas.html">are_mrca_taxa_names_in_fastas</a></td>
<td>Determine if the MRCA priors' taxa names are present in the FASTA files</td></tr>
<tr><td style="width: 25%;"><a href="are_mrca_taxa_non_intersecting.html">are_mrca_taxa_non_intersecting</a></td>
<td>Determine if MRCA priors that are monophyletic (i.e., assume no other taxons share a same MRCA) are compatible, that is, there are no taxa that intersect. For example, one MRCA prior with taxon 1 and 2 is incompatible with an MRCA prior with taxon 3 and 4</td></tr>
<tr><td style="width: 25%;"><a href="are_rln_clock_models.html">are_rln_clock_models</a></td>
<td>Are the clock models Relaxed Log-Normal clock models?</td></tr>
<tr><td style="width: 25%;"><a href="are_site_models.html">are_site_models</a></td>
<td>Determine if x consists out of site_models objects</td></tr>
<tr><td style="width: 25%;"><a href="are_tree_priors.html">are_tree_priors</a></td>
<td>Determine if x consists out of tree_priors objects</td></tr>
</table>

<h2><a name="B">-- B --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="bd_tree_prior_to_xml_prior_distr.html">bd_tree_prior_to_xml_prior_distr</a></td>
<td>Creates the tree prior section in the prior section of the prior section of the distribution section of a BEAST2 XML parameter file for a Birth-Death tree prior</td></tr>
</table>

<h2><a name="C">-- C --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="cbs_tree_prior_to_xml_prior_distr.html">cbs_tree_prior_to_xml_prior_distr</a></td>
<td>Creates the tree prior section in the prior section of the prior section of the distribution section of a BEAST2 XML parameter file for a Birth-Death tree prior</td></tr>
<tr><td style="width: 25%;"><a href="ccp_tree_prior_to_xml_prior_distr.html">ccp_tree_prior_to_xml_prior_distr</a></td>
<td>Creates the tree prior section in the prior section of the prior section of the distribution section of a BEAST2 XML parameter file for a Coalescent Constant Population tree prior</td></tr>
<tr><td style="width: 25%;"><a href="cep_tree_prior_to_xml_prior_distr.html">cep_tree_prior_to_xml_prior_distr</a></td>
<td>Creates the tree prior section in the prior section of the prior section of the distribution section of a BEAST2 XML parameter file for a Coalescent Exponential Population tree prior</td></tr>
<tr><td style="width: 25%;"><a href="clock_models_to_xml_operators.html">clock_models_to_xml_operators</a></td>
<td>Create all clock models' operators' XML text</td></tr>
<tr><td style="width: 25%;"><a href="clock_models_to_xml_prior_distr.html">clock_models_to_xml_prior_distr</a></td>
<td>Represent the clock models as XML</td></tr>
<tr><td style="width: 25%;"><a href="clock_models_to_xml_state.html">clock_models_to_xml_state</a></td>
<td>Converts one or more clock models to the 'state' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="clock_models_to_xml_tracelog.html">clock_models_to_xml_tracelog</a></td>
<td>Creates the clock models' XML for the tracelog section</td></tr>
<tr><td style="width: 25%;"><a href="clock_model_to_xml_lh_distr.html">clock_model_to_xml_lh_distr</a></td>
<td>Converts a clock model to the 'branchRateModel' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="clock_model_to_xml_operators.html">clock_model_to_xml_operators</a></td>
<td>Converts a clock model to the 'operators' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="clock_model_to_xml_prior_distr.html">clock_model_to_xml_prior_distr</a></td>
<td>Converts a clock model to the 'prior' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="clock_model_to_xml_state.html">clock_model_to_xml_state</a></td>
<td>Converts a clock model to the 'state' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="clock_model_to_xml_tracelog.html">clock_model_to_xml_tracelog</a></td>
<td>Creates the clock model's XML for the tracelog section</td></tr>
<tr><td style="width: 25%;"><a href="compare_lines.html">compare_lines</a></td>
<td>Internal debug function to compare the actually created lines to expected lines.</td></tr>
<tr><td style="width: 25%;"><a href="count_trailing_spaces.html">count_trailing_spaces</a></td>
<td>Count the number of spaces before the first character</td></tr>
<tr><td style="width: 25%;"><a href="create_alpha_param.html">create_alpha_param</a></td>
<td>Create a parameter called alpha</td></tr>
<tr><td style="width: 25%;"><a href="create_bd_tree_prior.html">create_bd_tree_prior</a></td>
<td>Create a Birth-Death tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input.html">create_beast2_input</a></td>
<td>Create a BEAST2 XML input text</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_1_12.html">create_beast2_input_1_12</a></td>
<td>Create a BEAST2 XML input text, interface of v1.12</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_beast.html">create_beast2_input_beast</a></td>
<td>Creates the beast section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_data.html">create_beast2_input_data</a></td>
<td>Creates the data section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_data_sequences.html">create_beast2_input_data_sequences</a></td>
<td>Creates the data section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_distr.html">create_beast2_input_distr</a></td>
<td>Creates the distribution section of a BEAST2 XML parameter file.</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_distr_lh.html">create_beast2_input_distr_lh</a></td>
<td>Creates the likelihood section in the distribution section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_distr_prior.html">create_beast2_input_distr_prior</a></td>
<td>Creates the prior section in the distribution section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_file.html">create_beast2_input_file</a></td>
<td>Create a BEAST2 input file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_file_1_12.html">create_beast2_input_file_1_12</a></td>
<td>Create a BEAST2 input file, interface of v1.12. This interface is obsoleted, use 'create_beast2_input_file' instead</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_init.html">create_beast2_input_init</a></td>
<td>Creates the map section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_loggers.html">create_beast2_input_loggers</a></td>
<td>Creates the two logger sections of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_map.html">create_beast2_input_map</a></td>
<td>Creates the map section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_operators.html">create_beast2_input_operators</a></td>
<td>Creates the operators section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_run.html">create_beast2_input_run</a></td>
<td>Creates the state section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_screenlog.html">create_beast2_input_screenlog</a></td>
<td>Creates the screenlog section of the logger section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_state.html">create_beast2_input_state</a></td>
<td>Creates the state section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_tracelog.html">create_beast2_input_tracelog</a></td>
<td>Creates the tracelog section of the logger section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_treelogs.html">create_beast2_input_treelogs</a></td>
<td>Creates the tracelog section of the logger section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beast2_input_xml.html">create_beast2_input_xml</a></td>
<td>Creates the xml section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="create_beta_distr.html">create_beta_distr</a></td>
<td>Create a beta distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_beta_param.html">create_beta_param</a></td>
<td>Create a parameter called beta</td></tr>
<tr><td style="width: 25%;"><a href="create_cbs_tree_prior.html">create_cbs_tree_prior</a></td>
<td>Create a Coalescent Bayesian Skyline tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_ccp_tree_prior.html">create_ccp_tree_prior</a></td>
<td>Create a Coalescent Constant Population tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_cep_tree_prior.html">create_cep_tree_prior</a></td>
<td>Create a Coalescent Exponential Population tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_clock_model.html">create_clock_model</a></td>
<td>General function to create a clock model</td></tr>
<tr><td style="width: 25%;"><a href="create_clock_models.html">create_clock_models</a></td>
<td>Creates all supported clock models, which is just a list of the types returned by 'create_rln_clock_model', and 'create_strict_clock_model'</td></tr>
<tr><td style="width: 25%;"><a href="create_rln_clock_model.html">create_clock_model_rln</a></td>
<td>Create a relaxed log-normal clock model</td></tr>
<tr><td style="width: 25%;"><a href="create_strict_clock_model.html">create_clock_model_strict</a></td>
<td>Create a strict clock model</td></tr>
<tr><td style="width: 25%;"><a href="create_clock_rate_param.html">create_clock_rate_param</a></td>
<td>Create a parameter called 'clock_rate', as needed by 'create_strict_clock_model'</td></tr>
<tr><td style="width: 25%;"><a href="create_distr.html">create_distr</a></td>
<td>General function to create a distribution.</td></tr>
<tr><td style="width: 25%;"><a href="create_beta_distr.html">create_distr_beta</a></td>
<td>Create a beta distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_exp_distr.html">create_distr_exp</a></td>
<td>Create an exponential distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_gamma_distr.html">create_distr_gamma</a></td>
<td>Create a gamma distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_inv_gamma_distr.html">create_distr_inv_gamma</a></td>
<td>Create an inverse gamma distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_laplace_distr.html">create_distr_laplace</a></td>
<td>Create a Laplace distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_log_normal_distr.html">create_distr_log_normal</a></td>
<td>Create a log-normal distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_normal_distr.html">create_distr_normal</a></td>
<td>Create an normal distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_one_div_x_distr.html">create_distr_one_div_x</a></td>
<td>Create a 1/x distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_poisson_distr.html">create_distr_poisson</a></td>
<td>Create a Poisson distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_uniform_distr.html">create_distr_uniform</a></td>
<td>Create a uniform distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_exp_distr.html">create_exp_distr</a></td>
<td>Create an exponential distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_gamma_distr.html">create_gamma_distr</a></td>
<td>Create a gamma distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_gamma_site_model.html">create_gamma_site_model</a></td>
<td>Create a gamma site model, part of a site model</td></tr>
<tr><td style="width: 25%;"><a href="create_gtr_site_model.html">create_gtr_site_model</a></td>
<td>Create a GTR site model</td></tr>
<tr><td style="width: 25%;"><a href="create_hky_site_model.html">create_hky_site_model</a></td>
<td>Create an HKY site model</td></tr>
<tr><td style="width: 25%;"><a href="create_inv_gamma_distr.html">create_inv_gamma_distr</a></td>
<td>Create an inverse gamma distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_jc69_site_model.html">create_jc69_site_model</a></td>
<td>Create a JC69 site model</td></tr>
<tr><td style="width: 25%;"><a href="create_jc69_site_models.html">create_jc69_site_models</a></td>
<td>Creates a JC69_site_model for each ID</td></tr>
<tr><td style="width: 25%;"><a href="create_kappa_1_param.html">create_kappa_1_param</a></td>
<td>Create a parameter called kappa 1</td></tr>
<tr><td style="width: 25%;"><a href="create_kappa_2_param.html">create_kappa_2_param</a></td>
<td>Create a parameter called kappa 2</td></tr>
<tr><td style="width: 25%;"><a href="create_lambda_param.html">create_lambda_param</a></td>
<td>Create a parameter called lambda</td></tr>
<tr><td style="width: 25%;"><a href="create_laplace_distr.html">create_laplace_distr</a></td>
<td>Create a Laplace distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_log_normal_distr.html">create_log_normal_distr</a></td>
<td>Create a log-normal distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_mcmc.html">create_mcmc</a></td>
<td>Function to create the MCMC options, as in the BEAUti MCMC tab.</td></tr>
<tr><td style="width: 25%;"><a href="create_mean_param.html">create_mean_param</a></td>
<td>Create a parameter called mean</td></tr>
<tr><td style="width: 25%;"><a href="create_misc_options.html">create_misc_options</a></td>
<td>Function to create a misc_options, containing miscellaneous options to fine-tune the created BEAST2 XML file. Whatever option chosen here, the created XML file will be valid.</td></tr>
<tr><td style="width: 25%;"><a href="create_mrca_prior.html">create_mrca_prior</a></td>
<td>Create a Most Recent Common Ancestor prior</td></tr>
<tr><td style="width: 25%;"><a href="create_mu_param.html">create_mu_param</a></td>
<td>Create a parameter called mu</td></tr>
<tr><td style="width: 25%;"><a href="create_m_param.html">create_m_param</a></td>
<td>Create a parameter called m</td></tr>
<tr><td style="width: 25%;"><a href="create_normal_distr.html">create_normal_distr</a></td>
<td>Create an normal distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_one_div_x_distr.html">create_one_div_x_distr</a></td>
<td>Create a 1/x distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_param.html">create_param</a></td>
<td>General function to create a parameter.</td></tr>
<tr><td style="width: 25%;"><a href="create_alpha_param.html">create_param_alpha</a></td>
<td>Create a parameter called alpha</td></tr>
<tr><td style="width: 25%;"><a href="create_beta_param.html">create_param_beta</a></td>
<td>Create a parameter called beta</td></tr>
<tr><td style="width: 25%;"><a href="create_clock_rate_param.html">create_param_clock_rate</a></td>
<td>Create a parameter called 'clock_rate', as needed by 'create_strict_clock_model'</td></tr>
<tr><td style="width: 25%;"><a href="create_kappa_1_param.html">create_param_kappa_1</a></td>
<td>Create a parameter called kappa 1</td></tr>
<tr><td style="width: 25%;"><a href="create_kappa_2_param.html">create_param_kappa_2</a></td>
<td>Create a parameter called kappa 2</td></tr>
<tr><td style="width: 25%;"><a href="create_lambda_param.html">create_param_lambda</a></td>
<td>Create a parameter called lambda</td></tr>
<tr><td style="width: 25%;"><a href="create_m_param.html">create_param_m</a></td>
<td>Create a parameter called m</td></tr>
<tr><td style="width: 25%;"><a href="create_mean_param.html">create_param_mean</a></td>
<td>Create a parameter called mean</td></tr>
<tr><td style="width: 25%;"><a href="create_mu_param.html">create_param_mu</a></td>
<td>Create a parameter called mu</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_ac_param.html">create_param_rate_ac</a></td>
<td>Create a parameter called 'rate AC'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_ag_param.html">create_param_rate_ag</a></td>
<td>Create a parameter called 'rate AG'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_at_param.html">create_param_rate_at</a></td>
<td>Create a parameter called 'rate AT'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_cg_param.html">create_param_rate_cg</a></td>
<td>Create a parameter called 'rate CG'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_ct_param.html">create_param_rate_ct</a></td>
<td>Create a parameter called 'rate CT'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_gt_param.html">create_param_rate_gt</a></td>
<td>Create a parameter called 'rate GT'</td></tr>
<tr><td style="width: 25%;"><a href="create_s_param.html">create_param_s</a></td>
<td>Create a parameter called s</td></tr>
<tr><td style="width: 25%;"><a href="create_scale_param.html">create_param_scale</a></td>
<td>Create a parameter called scale</td></tr>
<tr><td style="width: 25%;"><a href="create_sigma_param.html">create_param_sigma</a></td>
<td>Create a parameter called sigma</td></tr>
<tr><td style="width: 25%;"><a href="create_poisson_distr.html">create_poisson_distr</a></td>
<td>Create a Poisson distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_random_name.html">create_random_name</a></td>
<td>Create a random name, for example, to be used by 'create_mrca_prior'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_ac_param.html">create_rate_ac_param</a></td>
<td>Create a parameter called 'rate AC'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_ag_param.html">create_rate_ag_param</a></td>
<td>Create a parameter called 'rate AG'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_at_param.html">create_rate_at_param</a></td>
<td>Create a parameter called 'rate AT'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_cg_param.html">create_rate_cg_param</a></td>
<td>Create a parameter called 'rate CG'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_ct_param.html">create_rate_ct_param</a></td>
<td>Create a parameter called 'rate CT'</td></tr>
<tr><td style="width: 25%;"><a href="create_rate_gt_param.html">create_rate_gt_param</a></td>
<td>Create a parameter called 'rate GT'</td></tr>
<tr><td style="width: 25%;"><a href="create_rln_clock_model.html">create_rln_clock_model</a></td>
<td>Create a relaxed log-normal clock model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_alpha_param.html">create_rnd_alpha_param</a></td>
<td>Create a random alpha parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_bd_tree_prior.html">create_rnd_bd_tree_prior</a></td>
<td>Create a random BD tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_beta_distr.html">create_rnd_beta_distr</a></td>
<td>Create a random beta distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_beta_param.html">create_rnd_beta_param</a></td>
<td>Create a random beta parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_bool.html">create_rnd_bool</a></td>
<td>Create a random boolean</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_cbs_tree_prior.html">create_rnd_cbs_tree_prior</a></td>
<td>Create a random CBS tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_ccp_tree_prior.html">create_rnd_ccp_tree_prior</a></td>
<td>Create a random CCP tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_cep_tree_prior.html">create_rnd_cep_tree_prior</a></td>
<td>Create a random CEP tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_clock_model.html">create_rnd_clock_model</a></td>
<td>Create a random clock model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_clock_rate_param.html">create_rnd_clock_rate_param</a></td>
<td>Create a random clock rate parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_distr.html">create_rnd_distr</a></td>
<td>Create a random distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_estimate.html">create_rnd_estimate</a></td>
<td>Create a random value for 'estimate', which must be TRUE or FALSE</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_exp_distr.html">create_rnd_exp_distr</a></td>
<td>Create a random exponential distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_freq_equilibrium.html">create_rnd_freq_equilibrium</a></td>
<td>Create a random value for 'freq equilibrium', which can be 'estimated', 'empirical' or 'all_equal'</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_gamma_distr.html">create_rnd_gamma_distr</a></td>
<td>Create a random gamma distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_gamma_site_model.html">create_rnd_gamma_site_model</a></td>
<td>Create a random gamma site model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_gtr_site_model.html">create_rnd_gtr_site_model</a></td>
<td>Create a random GTR site model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_hky_site_model.html">create_rnd_hky_site_model</a></td>
<td>Create a random HKY site model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_inv_gamma_distr.html">create_rnd_inv_gamma_distr</a></td>
<td>Create a random inverse-gamma distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_jc69_site_model.html">create_rnd_jc69_site_model</a></td>
<td>Create a random JC69 distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_kappa_1_param.html">create_rnd_kappa_1_param</a></td>
<td>Create a random kappa 1 parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_kappa_2_param.html">create_rnd_kappa_2_param</a></td>
<td>Create a random kappa 2 parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_lambda_param.html">create_rnd_lambda_param</a></td>
<td>Create a random lambda parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_laplace_distr.html">create_rnd_laplace_distr</a></td>
<td>Create a random Laplace distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_log_normal_distr.html">create_rnd_log_normal_distr</a></td>
<td>Create a random log-normal distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_mean_param.html">create_rnd_mean_param</a></td>
<td>Create a random mean parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_mrca_prior.html">create_rnd_mrca_prior</a></td>
<td>Create a random MRCA prior</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_mrca_priors.html">create_rnd_mrca_priors</a></td>
<td>Creates a random MRCA prior option. This is either zero, one or more MRCA priors. For zero MRCA priors, NA is used. When two or more MRCA priors (see create_rnd_two_mrca_priors) are created, those are checked to be compatible</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_mu_param.html">create_rnd_mu_param</a></td>
<td>Create a random mu parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_m_param.html">create_rnd_m_param</a></td>
<td>Create a random m parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_normal_distr.html">create_rnd_normal_distr</a></td>
<td>Create a random normal distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_one_div_x_distr.html">create_rnd_one_div_x_distr</a></td>
<td>Create a random 1/x distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_param.html">create_rnd_param</a></td>
<td>Create a random parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_poisson_distr.html">create_rnd_poisson_distr</a></td>
<td>Create a random Poisson distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_rate_ac_param.html">create_rnd_rate_ac_param</a></td>
<td>Create a random rate AC parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_rate_ag_param.html">create_rnd_rate_ag_param</a></td>
<td>Create a random rate AG parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_rate_at_param.html">create_rnd_rate_at_param</a></td>
<td>Create a random rate AT parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_rate_cg_param.html">create_rnd_rate_cg_param</a></td>
<td>Create a random rate CG parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_rate_ct_param.html">create_rnd_rate_ct_param</a></td>
<td>Create a random rate CT parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_rate_gt_param.html">create_rnd_rate_gt_param</a></td>
<td>Create a random rate GT parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_rln_clock_model.html">create_rnd_rln_clock_model</a></td>
<td>Create a random RLN clock model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_scale_param.html">create_rnd_scale_param</a></td>
<td>Create a random scale parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_sigma_param.html">create_rnd_sigma_param</a></td>
<td>Create a random sigma parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_site_model.html">create_rnd_site_model</a></td>
<td>Create a random site model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_strict_clock_model.html">create_rnd_strict_clock_model</a></td>
<td>Create a random strict clock model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_s_param.html">create_rnd_s_param</a></td>
<td>Create a random s parameter</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_tn93_site_model.html">create_rnd_tn93_site_model</a></td>
<td>Create a random TN93 site model</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_tree_prior.html">create_rnd_tree_prior</a></td>
<td>Create a random tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_two_mrca_priors.html">create_rnd_two_mrca_priors</a></td>
<td>Creates two MRCA priors, checked to be compatible</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_uniform_distr.html">create_rnd_uniform_distr</a></td>
<td>Create a random uniform distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_rnd_yule_tree_prior.html">create_rnd_yule_tree_prior</a></td>
<td>Create a random Yule tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_scale_param.html">create_scale_param</a></td>
<td>Create a parameter called scale</td></tr>
<tr><td style="width: 25%;"><a href="create_sigma_param.html">create_sigma_param</a></td>
<td>Create a parameter called sigma</td></tr>
<tr><td style="width: 25%;"><a href="create_site_model.html">create_site_model</a></td>
<td>General function to create a site model.</td></tr>
<tr><td style="width: 25%;"><a href="create_site_models.html">create_site_models</a></td>
<td>Creates all supported site models which is just a list of the types returned by 'create_gtr_site_model', 'create_hky_site_model', 'create_jc69_site_model' and 'create_tn93_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="create_gtr_site_model.html">create_site_model_gtr</a></td>
<td>Create a GTR site model</td></tr>
<tr><td style="width: 25%;"><a href="create_hky_site_model.html">create_site_model_hky</a></td>
<td>Create an HKY site model</td></tr>
<tr><td style="width: 25%;"><a href="create_jc69_site_model.html">create_site_model_jc69</a></td>
<td>Create a JC69 site model</td></tr>
<tr><td style="width: 25%;"><a href="create_tn93_site_model.html">create_site_model_tn93</a></td>
<td>Create a TN93 site model</td></tr>
<tr><td style="width: 25%;"><a href="create_strict_clock_model.html">create_strict_clock_model</a></td>
<td>Create a strict clock model</td></tr>
<tr><td style="width: 25%;"><a href="create_strict_clock_models.html">create_strict_clock_models</a></td>
<td>Creates n strict clock_models</td></tr>
<tr><td style="width: 25%;"><a href="create_s_param.html">create_s_param</a></td>
<td>Create a parameter called s</td></tr>
<tr><td style="width: 25%;"><a href="create_tn93_site_model.html">create_tn93_site_model</a></td>
<td>Create a TN93 site model</td></tr>
<tr><td style="width: 25%;"><a href="create_tree_prior.html">create_tree_prior</a></td>
<td>Internal function to create a tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_tree_priors.html">create_tree_priors</a></td>
<td>Creates all supported tree priors, which is just a list of the types returned by 'create_bd_tree_prior', 'create_cbs_tree_prior', 'create_ccp_tree_prior', 'create_cep_tree_prior' and 'create_yule_tree_prior'</td></tr>
<tr><td style="width: 25%;"><a href="create_bd_tree_prior.html">create_tree_prior_bd</a></td>
<td>Create a Birth-Death tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_cbs_tree_prior.html">create_tree_prior_cbs</a></td>
<td>Create a Coalescent Bayesian Skyline tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_ccp_tree_prior.html">create_tree_prior_ccp</a></td>
<td>Create a Coalescent Constant Population tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_cep_tree_prior.html">create_tree_prior_cep</a></td>
<td>Create a Coalescent Exponential Population tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_yule_tree_prior.html">create_tree_prior_yule</a></td>
<td>Create a Yule tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_uniform_distr.html">create_uniform_distr</a></td>
<td>Create a uniform distribution</td></tr>
<tr><td style="width: 25%;"><a href="create_yule_tree_prior.html">create_yule_tree_prior</a></td>
<td>Create a Yule tree prior</td></tr>
<tr><td style="width: 25%;"><a href="create_yule_tree_priors.html">create_yule_tree_priors</a></td>
<td>Creates n Yule tree priors</td></tr>
</table>

<h2><a name="D">-- D --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="default_params_doc.html">default_params_doc</a></td>
<td>This function does nothing. It is intended to inherit is parameters' documentation.</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml.html">distr_to_xml</a></td>
<td>Converts a distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_beta.html">distr_to_xml_beta</a></td>
<td>Converts a beta distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_exp.html">distr_to_xml_exp</a></td>
<td>Converts an exponential distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_gamma.html">distr_to_xml_gamma</a></td>
<td>Converts a gamma distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_inv_gamma.html">distr_to_xml_inv_gamma</a></td>
<td>Converts a inv_gamma distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_laplace.html">distr_to_xml_laplace</a></td>
<td>Converts a laplace distibution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_log_normal.html">distr_to_xml_log_normal</a></td>
<td>Converts a log-normal distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_normal.html">distr_to_xml_normal</a></td>
<td>Converts a normal distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_one_div_x.html">distr_to_xml_one_div_x</a></td>
<td>Converts a 1/x distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_poisson.html">distr_to_xml_poisson</a></td>
<td>Converts a poisson distribution to XML</td></tr>
<tr><td style="width: 25%;"><a href="distr_to_xml_uniform.html">distr_to_xml_uniform</a></td>
<td>Converts a uniform distribution to XML</td></tr>
</table>

<h2><a name="E">-- E --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="extract_xml_loggers_from_lines.html">extract_xml_loggers_from_lines</a></td>
<td>Extract everything between first loggers and last loggers line</td></tr>
<tr><td style="width: 25%;"><a href="extract_xml_operators_from_lines.html">extract_xml_operators_from_lines</a></td>
<td>Extract everything between first operators and last operators line</td></tr>
<tr><td style="width: 25%;"><a href="extract_xml_section_from_lines.html">extract_xml_section_from_lines</a></td>
<td>Get the lines of an XML section, including the section tags</td></tr>
</table>

<h2><a name="F">-- F --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="fastas_to_phylos.html">fastas_to_phylos</a></td>
<td>Create a random phylogeny, with the same taxa names as the FASTA file and the desired crown age</td></tr>
<tr><td style="width: 25%;"><a href="fasta_file_to_sequences.html">fasta_file_to_sequences</a></td>
<td>Convert a FASTA file to a table of sequences</td></tr>
<tr><td style="width: 25%;"><a href="fasta_to_phylo.html">fasta_to_phylo</a></td>
<td>Create a random phylogeny, with the same taxa names as the FASTA file and the desired crown age</td></tr>
<tr><td style="width: 25%;"><a href="files_exist.html">files_exist</a></td>
<td>Multi-file file.exists function</td></tr>
<tr><td style="width: 25%;"><a href="find_clock_model.html">find_clock_model</a></td>
<td>Finds a clock model with a certain ID</td></tr>
<tr><td style="width: 25%;"><a href="find_first_regex_line.html">find_first_regex_line</a></td>
<td>Find the first line that satisfies a regex</td></tr>
<tr><td style="width: 25%;"><a href="find_first_xml_opening_tag_line.html">find_first_xml_opening_tag_line</a></td>
<td>Find the line number of the first section's opening tag</td></tr>
<tr><td style="width: 25%;"><a href="find_last_xml_closing_tag_line.html">find_last_xml_closing_tag_line</a></td>
<td>Find the highest line number of a section's closing tag</td></tr>
<tr><td style="width: 25%;"><a href="freq_equilibrium_to_xml.html">freq_equilibrium_to_xml</a></td>
<td>Creates the 'freq_equilibrium' as XML</td></tr>
</table>

<h2><a name="G">-- G --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="gamma_site_models_to_xml_prior_distr.html">gamma_site_models_to_xml_prior_distr</a></td>
<td>Creates the gamma site models section in the distribution section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="gamma_site_model_to_xml_prior_distr.html">gamma_site_model_to_xml_prior_distr</a></td>
<td>Creates the gamma site models section in the distribution section of a BEAST2 XML parameter file</td></tr>
<tr><td style="width: 25%;"><a href="gamma_site_model_to_xml_state.html">gamma_site_model_to_xml_state</a></td>
<td>Converts a gamma site model to XML, used in the 'state' section</td></tr>
<tr><td style="width: 25%;"><a href="get_alignment_id.html">get_alignment_id</a></td>
<td>Get the alignment ID from a FASTA filename</td></tr>
<tr><td style="width: 25%;"><a href="get_alignment_ids.html">get_alignment_ids</a></td>
<td>Get the alignment ID from one or more FASTA filenames</td></tr>
<tr><td style="width: 25%;"><a href="get_beautier_path.html">get_beautier_path</a></td>
<td>Get the full path of a file in the 'inst/extdata' folder</td></tr>
<tr><td style="width: 25%;"><a href="get_beautier_paths.html">get_beautier_paths</a></td>
<td>Get the full paths of files in the 'inst/extdata' folder</td></tr>
<tr><td style="width: 25%;"><a href="get_clock_models_ids.html">get_clock_models_ids</a></td>
<td>Collect the IDs of the list of clock models</td></tr>
<tr><td style="width: 25%;"><a href="get_clock_model_name.html">get_clock_model_name</a></td>
<td>Get the BEAUti name for a clock model</td></tr>
<tr><td style="width: 25%;"><a href="get_clock_model_names.html">get_clock_model_names</a></td>
<td>Get the clock model names</td></tr>
<tr><td style="width: 25%;"><a href="get_distr_names.html">get_distr_names</a></td>
<td>Get the distribution names</td></tr>
<tr><td style="width: 25%;"><a href="get_distr_n_params.html">get_distr_n_params</a></td>
<td>Get the number of parameters a distribution uses</td></tr>
<tr><td style="width: 25%;"><a href="get_fasta_filename.html">get_fasta_filename</a></td>
<td>Get the path of a FASTA file used in testing</td></tr>
<tr><td style="width: 25%;"><a href="get_file_base_sans_ext.html">get_file_base_sans_ext</a></td>
<td>Get the filename's base without extension</td></tr>
<tr><td style="width: 25%;"><a href="get_first_clock_model_index.html">get_first_clock_model_index</a></td>
<td>Get the index of a clock model with a list of clock models</td></tr>
<tr><td style="width: 25%;"><a href="get_freq_equilibrium_names.html">get_freq_equilibrium_names</a></td>
<td>Returns valid values for the 'freq_equilibrium' argument</td></tr>
<tr><td style="width: 25%;"><a href="get_has_non_strict_clock_model.html">get_has_non_strict_clock_model</a></td>
<td>Determines if there is at least one non-strict clock model in the list of one or more clock models</td></tr>
<tr><td style="width: 25%;"><a href="get_id.html">get_id</a></td>
<td>Conclude the ID from a FASTA filename</td></tr>
<tr><td style="width: 25%;"><a href="get_ids.html">get_ids</a></td>
<td>Conclude the IDs from one or more FASTA filenames</td></tr>
<tr><td style="width: 25%;"><a href="get_n_taxa.html">get_n_taxa</a></td>
<td>Extract the number of taxa from a file</td></tr>
<tr><td style="width: 25%;"><a href="get_operator_id_pre.html">get_operator_id_pre</a></td>
<td>Get the prefix of operator IDs</td></tr>
<tr><td style="width: 25%;"><a href="get_param_names.html">get_param_names</a></td>
<td>Get the parameter names</td></tr>
<tr><td style="width: 25%;"><a href="get_phylo_crown_age.html">get_phylo_crown_age</a></td>
<td>Obtain the crown age of a phylony</td></tr>
<tr><td style="width: 25%;"><a href="get_site_models_n_distrs.html">get_site_models_n_distrs</a></td>
<td>Get the number of distributions a site model has</td></tr>
<tr><td style="width: 25%;"><a href="get_site_models_n_params.html">get_site_models_n_params</a></td>
<td>Get the number of distributions one or more site models have</td></tr>
<tr><td style="width: 25%;"><a href="get_site_model_names.html">get_site_model_names</a></td>
<td>Get the site models' names</td></tr>
<tr><td style="width: 25%;"><a href="get_site_model_n_distrs.html">get_site_model_n_distrs</a></td>
<td>Get the number of distributions a site model has</td></tr>
<tr><td style="width: 25%;"><a href="get_site_model_n_params.html">get_site_model_n_params</a></td>
<td>Get the number of distributions a site model has</td></tr>
<tr><td style="width: 25%;"><a href="get_taxa_names.html">get_taxa_names</a></td>
<td>Extract the names of taxa from a file</td></tr>
<tr><td style="width: 25%;"><a href="get_tree_priors_n_distrs.html">get_tree_priors_n_distrs</a></td>
<td>Get the number of distributions a tree prior has</td></tr>
<tr><td style="width: 25%;"><a href="get_tree_priors_n_params.html">get_tree_priors_n_params</a></td>
<td>Get the number of parameters a list of tree priors has</td></tr>
<tr><td style="width: 25%;"><a href="get_tree_prior_names.html">get_tree_prior_names</a></td>
<td>Get the tree prior names</td></tr>
<tr><td style="width: 25%;"><a href="get_tree_prior_n_distrs.html">get_tree_prior_n_distrs</a></td>
<td>Get the number of distributions a tree prior has</td></tr>
<tr><td style="width: 25%;"><a href="get_tree_prior_n_params.html">get_tree_prior_n_params</a></td>
<td>Get the number of parameters a tree prior has</td></tr>
<tr><td style="width: 25%;"><a href="get_unlinked_clock_models.html">get_unlinked_clock_models</a></td>
<td>Get the first clock model of each ID</td></tr>
<tr><td style="width: 25%;"><a href="get_unlinked_site_models.html">get_unlinked_site_models</a></td>
<td>Get the first site model of each ID</td></tr>
<tr><td style="width: 25%;"><a href="get_unlinked_tree_priors.html">get_unlinked_tree_priors</a></td>
<td>Get the first tree prior of each ID</td></tr>
<tr><td style="width: 25%;"><a href="get_xml_closing_tag.html">get_xml_closing_tag</a></td>
<td>Get the XML closing tag</td></tr>
<tr><td style="width: 25%;"><a href="get_xml_opening_tag.html">get_xml_opening_tag</a></td>
<td>Get the XML opening tag</td></tr>
</table>

<h2><a name="H">-- H --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="has_shared_rln_clock_models.html">has_shared_rln_clock_models</a></td>
<td>Are there any shared RLN clock models?</td></tr>
<tr><td style="width: 25%;"><a href="has_xml_closing_tag.html">has_xml_closing_tag</a></td>
<td>Is an XML opening tag with value 'section' present amongst the lines of the text?</td></tr>
<tr><td style="width: 25%;"><a href="has_xml_opening_tag.html">has_xml_opening_tag</a></td>
<td>Is an XML opening tag with value 'section' present amongst the lines of the text?</td></tr>
<tr><td style="width: 25%;"><a href="has_xml_short_closing_tag.html">has_xml_short_closing_tag</a></td>
<td>Is an XML closing tag with short closing text at the end of the text?</td></tr>
</table>

<h2><a name="I">-- I --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="indent.html">indent</a></td>
<td>Indent text for a certain number of spaces If the text is only whitespace, leave it as such</td></tr>
<tr><td style="width: 25%;"><a href="init_bd_tree_prior.html">init_bd_tree_prior</a></td>
<td>Initializes a Birth-Death tree prior</td></tr>
<tr><td style="width: 25%;"><a href="init_ccp_tree_prior.html">init_ccp_tree_prior</a></td>
<td>Initializes a Coalescent Constant Population tree prior</td></tr>
<tr><td style="width: 25%;"><a href="init_cep_tree_prior.html">init_cep_tree_prior</a></td>
<td>Initializes a Coalescent Exponential Population tree prior</td></tr>
<tr><td style="width: 25%;"><a href="init_clock_models.html">init_clock_models</a></td>
<td>Initializes all clock models</td></tr>
<tr><td style="width: 25%;"><a href="init_distr.html">init_distr</a></td>
<td>Initializes a distribution</td></tr>
<tr><td style="width: 25%;"><a href="init_gamma_site_model.html">init_gamma_site_model</a></td>
<td>Initializes all site models</td></tr>
<tr><td style="width: 25%;"><a href="init_gtr_site_model.html">init_gtr_site_model</a></td>
<td>Initializes a GTR site model</td></tr>
<tr><td style="width: 25%;"><a href="init_hky_site_model.html">init_hky_site_model</a></td>
<td>Initializes an HKY site model</td></tr>
<tr><td style="width: 25%;"><a href="init_jc69_site_model.html">init_jc69_site_model</a></td>
<td>Initializes a JC69 site model</td></tr>
<tr><td style="width: 25%;"><a href="init_mrca_priors.html">init_mrca_priors</a></td>
<td>Initializes all mrca priors</td></tr>
<tr><td style="width: 25%;"><a href="init_param.html">init_param</a></td>
<td>Initializes a parameter</td></tr>
<tr><td style="width: 25%;"><a href="init_rln_clock_model.html">init_rln_clock_model</a></td>
<td>Initializes a Relaxed Log-Normal clock model</td></tr>
<tr><td style="width: 25%;"><a href="init_site_models.html">init_site_models</a></td>
<td>Initializes all site models</td></tr>
<tr><td style="width: 25%;"><a href="init_strict_clock_model.html">init_strict_clock_model</a></td>
<td>Initializes a strict clock model</td></tr>
<tr><td style="width: 25%;"><a href="init_tn93_site_model.html">init_tn93_site_model</a></td>
<td>Initializes a TN93 site model</td></tr>
<tr><td style="width: 25%;"><a href="init_tree_priors.html">init_tree_priors</a></td>
<td>Initializes all tree priors</td></tr>
<tr><td style="width: 25%;"><a href="init_yule_tree_prior.html">init_yule_tree_prior</a></td>
<td>Initializes a Yule tree prior</td></tr>
<tr><td style="width: 25%;"><a href="interspace.html">interspace</a></td>
<td>Puts spaces in between the lines</td></tr>
<tr><td style="width: 25%;"><a href="is_alpha_param.html">is_alpha_param</a></td>
<td>Determine if the object is a valid alpha parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_bd_tree_prior.html">is_bd_tree_prior</a></td>
<td>Determine if the object is a valid Birth Death tree prior as returned by 'create_bd_tree_prior'</td></tr>
<tr><td style="width: 25%;"><a href="is_beta_distr.html">is_beta_distr</a></td>
<td>Determine if the object is a valid beta distribution, as created by 'create_beta_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_beta_param.html">is_beta_param</a></td>
<td>Determine if the object is a valid beta parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_cbs_tree_prior.html">is_cbs_tree_prior</a></td>
<td>Determine if the object is a valid constant coalescent Bayesian skyline prior as returned by 'create_cbs_tree_prior'</td></tr>
<tr><td style="width: 25%;"><a href="is_ccp_tree_prior.html">is_ccp_tree_prior</a></td>
<td>Determine if the object is a valid constant coalescence population tree prior, as returned by 'create_ccp_tree_prior'</td></tr>
<tr><td style="width: 25%;"><a href="is_cep_tree_prior.html">is_cep_tree_prior</a></td>
<td>Determine if the object is a valid coalescent exponential population tree prior as returned by 'create_cep_tree_prior'</td></tr>
<tr><td style="width: 25%;"><a href="is_clock_model.html">is_clock_model</a></td>
<td>Determine if the object is a valid clock_model</td></tr>
<tr><td style="width: 25%;"><a href="is_clock_model_name.html">is_clock_model_name</a></td>
<td>Determines if the name is a valid clock model name</td></tr>
<tr><td style="width: 25%;"><a href="is_clock_rate_param.html">is_clock_rate_param</a></td>
<td>Determine if the object is a valid clock_rate parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_distr.html">is_distr</a></td>
<td>Determine if the object is a valid distribution</td></tr>
<tr><td style="width: 25%;"><a href="is_distr_name.html">is_distr_name</a></td>
<td>Determines if the name is a valid distribution name</td></tr>
<tr><td style="width: 25%;"><a href="is_exp_distr.html">is_exp_distr</a></td>
<td>Determine if the object is a valid exponential distribution as created by 'create_exp_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_freq_equilibrium_name.html">is_freq_equilibrium_name</a></td>
<td>Checks if 'name' is a valid 'freq_equilibrium' argument value</td></tr>
<tr><td style="width: 25%;"><a href="is_gamma_distr.html">is_gamma_distr</a></td>
<td>Determine if the object is a valid gamma distribution, as created by 'create_gamma_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_gamma_site_model.html">is_gamma_site_model</a></td>
<td>Is object x a gamma site model?</td></tr>
<tr><td style="width: 25%;"><a href="is_gtr_site_model.html">is_gtr_site_model</a></td>
<td>Determine if the object is a valid GTR site model, as created by 'create_gtr_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_hky_site_model.html">is_hky_site_model</a></td>
<td>Determine if the object is a valid HKY site model, as created by 'create_hky_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_id.html">is_id</a></td>
<td>Determine if the object is a valid ID</td></tr>
<tr><td style="width: 25%;"><a href="is_index_of_first_shared_clock_model.html">is_index_of_first_shared_clock_model</a></td>
<td>Is the index the first of a series of clock models that are shared?</td></tr>
<tr><td style="width: 25%;"><a href="is_index_of_non_first_shared_clock_model.html">is_index_of_non_first_shared_clock_model</a></td>
<td>Is the index the first of a series of clock models that are shared?</td></tr>
<tr><td style="width: 25%;"><a href="is_init_bd_tree_prior.html">is_init_bd_tree_prior</a></td>
<td>Determine if x is an initialized Birth-Death tree_prior object</td></tr>
<tr><td style="width: 25%;"><a href="is_init_beta_distr.html">is_init_beta_distr</a></td>
<td>Determine if x is an initialized beta distribution object as created by 'create_beta_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_cbs_tree_prior.html">is_init_cbs_tree_prior</a></td>
<td>Determine if x is an initialized Coalescent Bayesian Skyline tree_prior object</td></tr>
<tr><td style="width: 25%;"><a href="is_init_ccp_tree_prior.html">is_init_ccp_tree_prior</a></td>
<td>Determine if x is an initialized Coalescent Constant Population tree_prior object</td></tr>
<tr><td style="width: 25%;"><a href="is_init_cep_tree_prior.html">is_init_cep_tree_prior</a></td>
<td>Determine if x is an initialized Coalescent Exponential Population tree_prior object</td></tr>
<tr><td style="width: 25%;"><a href="is_init_clock_model.html">is_init_clock_model</a></td>
<td>Determine if x is an initialized clock_model object, as created by 'create_clock_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_distr.html">is_init_distr</a></td>
<td>Determine if x is an initialized distribution object as created by 'create_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_exp_distr.html">is_init_exp_distr</a></td>
<td>Determine if x is an initialized exponential distribution object as created by 'create_exp_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_gamma_distr.html">is_init_gamma_distr</a></td>
<td>Determine if x is an initialized gamma distribution object</td></tr>
<tr><td style="width: 25%;"><a href="is_init_gamma_site_model.html">is_init_gamma_site_model</a></td>
<td>Determine if x is an initialized gamma site model, as created by 'create_gamma_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_gtr_site_model.html">is_init_gtr_site_model</a></td>
<td>Determine if x is an initialized GTR site model as created by 'create_gtr_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_hky_site_model.html">is_init_hky_site_model</a></td>
<td>Determine if x is an initialized hky site model as created by 'create_hky_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_inv_gamma_distr.html">is_init_inv_gamma_distr</a></td>
<td>Determine if x is an initialized inv_gamma distribution object as created by 'create_inv_gamma_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_jc69_site_model.html">is_init_jc69_site_model</a></td>
<td>Determine if x is an initialized JC69 site model as created by 'create_jc69_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_laplace_distr.html">is_init_laplace_distr</a></td>
<td>Determine if x is an initialized laplace distribution object as created by 'create_laplace_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_log_normal_distr.html">is_init_log_normal_distr</a></td>
<td>Determine if x is an initialized log_normal distribution object as created by 'create_log_normal_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_mrca_prior.html">is_init_mrca_prior</a></td>
<td>Determine if x is an initialized mrca_prior objects</td></tr>
<tr><td style="width: 25%;"><a href="is_init_normal_distr.html">is_init_normal_distr</a></td>
<td>Determine if x is an initialized normal distribution object as created by 'create_normal_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_one_div_x_distr.html">is_init_one_div_x_distr</a></td>
<td>Determine if x is an initialized one_div_x distribution object as created by 'create_one_div_x_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_param.html">is_init_param</a></td>
<td>Determine if x is an initialized parameter, as created by 'create_param'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_poisson_distr.html">is_init_poisson_distr</a></td>
<td>Determine if x is an initialized poisson distribution object as created by 'create_poisson_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_rln_clock_model.html">is_init_rln_clock_model</a></td>
<td>Determine if x is an initialized relaxed log-normal clock_model object</td></tr>
<tr><td style="width: 25%;"><a href="is_init_site_model.html">is_init_site_model</a></td>
<td>Determine if x is an initialized site model, as created by 'create_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_strict_clock_model.html">is_init_strict_clock_model</a></td>
<td>Determine if x is an initialized strict clock_model object</td></tr>
<tr><td style="width: 25%;"><a href="is_init_tn93_site_model.html">is_init_tn93_site_model</a></td>
<td>Determine if x is an initialized tn93 site model as created by 'create_tn93_site_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_tree_prior.html">is_init_tree_prior</a></td>
<td>Determine if x is an initialized tree_prior objects</td></tr>
<tr><td style="width: 25%;"><a href="is_init_uniform_distr.html">is_init_uniform_distr</a></td>
<td>Determine if x is an initialized uniform distribution object as created by 'create_uniform_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_init_yule_tree_prior.html">is_init_yule_tree_prior</a></td>
<td>Determine if x is an initialized Yule tree_prior object</td></tr>
<tr><td style="width: 25%;"><a href="is_inv_gamma_distr.html">is_inv_gamma_distr</a></td>
<td>Determine if the object is a valid inv_gamma distribution, as created by 'create_inv_gamma_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_in_patterns.html">is_in_patterns</a></td>
<td>Create a random beta distribution</td></tr>
<tr><td style="width: 25%;"><a href="is_jc69_site_model.html">is_jc69_site_model</a></td>
<td>Determine if the object is a valid JC69 site model</td></tr>
<tr><td style="width: 25%;"><a href="is_kappa_1_param.html">is_kappa_1_param</a></td>
<td>Determine if the object is a valid kappa 1 parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_kappa_2_param.html">is_kappa_2_param</a></td>
<td>Determine if the object is a valid kappa 2 parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_lambda_param.html">is_lambda_param</a></td>
<td>Determine if the object is a valid lambda parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_laplace_distr.html">is_laplace_distr</a></td>
<td>Determine if the object is a valid laplace distribution, as created by 'create_laplace_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_log_normal_distr.html">is_log_normal_distr</a></td>
<td>Determine if the object is a valid log-normal distribution, as created by 'create_log_normal_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_mcmc.html">is_mcmc</a></td>
<td>Determine if the object is a valid mcmc object</td></tr>
<tr><td style="width: 25%;"><a href="is_mean_param.html">is_mean_param</a></td>
<td>Determine if the object is a valid mean parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_misc_options.html">is_misc_options</a></td>
<td>Determine if the object is a valid misc_options</td></tr>
<tr><td style="width: 25%;"><a href="is_mrca_prior.html">is_mrca_prior</a></td>
<td>Determine of the object is an MRCA prior, as returned by 'create_mrca_prior'</td></tr>
<tr><td style="width: 25%;"><a href="is_mu_param.html">is_mu_param</a></td>
<td>Determine if the object is a valid mu parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_m_param.html">is_m_param</a></td>
<td>Determine if the object is a valid m parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_normal_distr.html">is_normal_distr</a></td>
<td>Determine if the object is a valid normal distribution as created by 'create_normal_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_one_div_x_distr.html">is_one_div_x_distr</a></td>
<td>Determine if the object is a valid 1/x distribution, as created by 'create_one_div_x_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_one_na.html">is_one_na</a></td>
<td>Determines if x is one NA</td></tr>
<tr><td style="width: 25%;"><a href="is_param.html">is_param</a></td>
<td>Determine if the object is a valid parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_param_name.html">is_param_name</a></td>
<td>Determines if the name is a valid parameter name</td></tr>
<tr><td style="width: 25%;"><a href="is_phylo.html">is_phylo</a></td>
<td>Checks if the input is a phylogeny</td></tr>
<tr><td style="width: 25%;"><a href="is_poisson_distr.html">is_poisson_distr</a></td>
<td>Determine if the object is a valid poisson distribution as created by 'create_poisson_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_rate_ac_param.html">is_rate_ac_param</a></td>
<td>Determine if the object is a valid 'rate AC' parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_rate_ag_param.html">is_rate_ag_param</a></td>
<td>Determine if the object is a valid 'rate AG' parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_rate_at_param.html">is_rate_at_param</a></td>
<td>Determine if the object is a valid 'rate AT' parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_rate_cg_param.html">is_rate_cg_param</a></td>
<td>Determine if the object is a valid 'rate CG' parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_rate_ct_param.html">is_rate_ct_param</a></td>
<td>Determine if the object is a valid 'rate CT' parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_rate_gt_param.html">is_rate_gt_param</a></td>
<td>Determine if the object is a valid 'rate GT' parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_rln_clock_model.html">is_rln_clock_model</a></td>
<td>Determine if the object is a valid relaxed log normal clock model</td></tr>
<tr><td style="width: 25%;"><a href="is_scale_param.html">is_scale_param</a></td>
<td>Determine if the object is a valid scale parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_sigma_param.html">is_sigma_param</a></td>
<td>Determine if the object is a valid sigma parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_site_model.html">is_site_model</a></td>
<td>Determine if the object is a valid site_model</td></tr>
<tr><td style="width: 25%;"><a href="is_site_model_name.html">is_site_model_name</a></td>
<td>Determines if the name is a valid site_model name</td></tr>
<tr><td style="width: 25%;"><a href="is_strict_clock_model.html">is_strict_clock_model</a></td>
<td>Determine if the object is a valid strict clock model, as returned by 'create_strict_clock_model'</td></tr>
<tr><td style="width: 25%;"><a href="is_s_param.html">is_s_param</a></td>
<td>Determine if the object is a valid s parameter</td></tr>
<tr><td style="width: 25%;"><a href="is_tn93_site_model.html">is_tn93_site_model</a></td>
<td>Determine if the object is a valid TN93 site model,</td></tr>
<tr><td style="width: 25%;"><a href="is_tree_prior.html">is_tree_prior</a></td>
<td>Determine if an object is a valid tree prior</td></tr>
<tr><td style="width: 25%;"><a href="is_tree_prior_name.html">is_tree_prior_name</a></td>
<td>Determines if the name is a valid tree prior name</td></tr>
<tr><td style="width: 25%;"><a href="is_uniform_distr.html">is_uniform_distr</a></td>
<td>Determine if the object is a valid uniform distribution as created by 'create_uniform_distr'</td></tr>
<tr><td style="width: 25%;"><a href="is_xml.html">is_xml</a></td>
<td>Checks if the text is a valid XML node, that is, it has a opening and matching closing tag</td></tr>
<tr><td style="width: 25%;"><a href="is_yule_tree_prior.html">is_yule_tree_prior</a></td>
<td>Determine if the object is a valid Yule tree prior, as returned by 'create_yule_tree_prior'</td></tr>
</table>

<h2><a name="M">-- M --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="mcmc_to_xml_run.html">mcmc_to_xml_run</a></td>
<td>Converts an MCMC object to the run section's XML</td></tr>
<tr><td style="width: 25%;"><a href="mrca_priors_to_xml_operators.html">mrca_priors_to_xml_operators</a></td>
<td>Creates the XML of a list of one or more MRCA priors, as used in the 'operators' section</td></tr>
<tr><td style="width: 25%;"><a href="mrca_priors_to_xml_prior_distr.html">mrca_priors_to_xml_prior_distr</a></td>
<td>Creates the distribution section in the prior section of the distribution section of a BEAST2 XML parameter file. These lines start with '&lt;distribution id='</td></tr>
<tr><td style="width: 25%;"><a href="mrca_priors_to_xml_state.html">mrca_priors_to_xml_state</a></td>
<td>Converts one or more MRCA priors to the 'state' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="mrca_priors_to_xml_tracelog.html">mrca_priors_to_xml_tracelog</a></td>
<td>Creates the mrca priors' XML for the tracelog section</td></tr>
<tr><td style="width: 25%;"><a href="mrca_prior_to_xml_lh_distr.html">mrca_prior_to_xml_lh_distr</a></td>
<td>Converts an MRCA prior to the 'branchRateModel' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="mrca_prior_to_xml_operators.html">mrca_prior_to_xml_operators</a></td>
<td>Creates the XML of an MRCA prior, as used in the 'operators' section</td></tr>
<tr><td style="width: 25%;"><a href="mrca_prior_to_xml_prior_distr.html">mrca_prior_to_xml_prior_distr</a></td>
<td>Creates the distribution section in the prior section of the distribution section of a BEAST2 XML parameter file. These lines start with '&lt;distribution id='</td></tr>
<tr><td style="width: 25%;"><a href="mrca_prior_to_xml_state.html">mrca_prior_to_xml_state</a></td>
<td>Creates the XML of an MRCA prior, as used in the 'state' section</td></tr>
<tr><td style="width: 25%;"><a href="mrca_prior_to_xml_taxonset.html">mrca_prior_to_xml_taxonset</a></td>
<td>Creates the distribution section in the prior section of the distribution section of a BEAST2 XML parameter file. These lines start with '&lt;distribution id='</td></tr>
<tr><td style="width: 25%;"><a href="mrca_prior_to_xml_tracelog.html">mrca_prior_to_xml_tracelog</a></td>
<td>Creates the mrca prior's XML for the tracelog section</td></tr>
</table>

<h2><a name="P">-- P --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="parameter_to_xml.html">parameter_to_xml</a></td>
<td>Converts a parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_alpha.html">parameter_to_xml_alpha</a></td>
<td>Converts an alpha parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_beta.html">parameter_to_xml_beta</a></td>
<td>Converts a beta parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_clock_rate.html">parameter_to_xml_clock_rate</a></td>
<td>Converts a 'clockRate' parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_kappa_1.html">parameter_to_xml_kappa_1</a></td>
<td>Converts a kappa 1 parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_kappa_2.html">parameter_to_xml_kappa_2</a></td>
<td>Converts a kappa 2 parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_lambda.html">parameter_to_xml_lambda</a></td>
<td>Converts a lambda parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_m.html">parameter_to_xml_m</a></td>
<td>Converts a m parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_mean.html">parameter_to_xml_mean</a></td>
<td>Converts a mean parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_mu.html">parameter_to_xml_mu</a></td>
<td>Converts a mu parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_rate_ac.html">parameter_to_xml_rate_ac</a></td>
<td>Converts a 'rate AC' parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_rate_ag.html">parameter_to_xml_rate_ag</a></td>
<td>Converts a 'rate AG' parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_rate_at.html">parameter_to_xml_rate_at</a></td>
<td>Converts a 'rate AT' parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_rate_cg.html">parameter_to_xml_rate_cg</a></td>
<td>Converts a 'rate CG' parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_rate_ct.html">parameter_to_xml_rate_ct</a></td>
<td>Converts a 'rate CT' parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_rate_gt.html">parameter_to_xml_rate_gt</a></td>
<td>Converts a 'rate GT' parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_s.html">parameter_to_xml_s</a></td>
<td>Converts a s parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_scale.html">parameter_to_xml_scale</a></td>
<td>Converts a scale parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="parameter_to_xml_sigma.html">parameter_to_xml_sigma</a></td>
<td>Converts a sigma parameter to XML</td></tr>
<tr><td style="width: 25%;"><a href="phylo_to_xml_state.html">phylo_to_xml_state</a></td>
<td>Creates the XML of a phylogeny, as used in the 'state' section</td></tr>
</table>

<h2><a name="R">-- R --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="remove_multiline.html">remove_multiline</a></td>
<td>Remove consequetive lines</td></tr>
<tr><td style="width: 25%;"><a href="rln_clock_model_to_xml_mean_rate_prior.html">rln_clock_model_to_xml_mean_rate_prior</a></td>
<td>Used by 'clock_models_to_xml_prior_distr'</td></tr>
<tr><td style="width: 25%;"><a href="rnd_phylo_to_xml_init.html">rnd_phylo_to_xml_init</a></td>
<td>Creates the XML of a random phylogeny, as used in the 'init' section</td></tr>
<tr><td style="width: 25%;"><a href="rnd_phylo_to_xml_state.html">rnd_phylo_to_xml_state</a></td>
<td>Creates the XML of a random phylogeny, as used in the 'state' section</td></tr>
</table>

<h2><a name="S">-- S --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="site_models_to_xml_prior_distr.html">site_models_to_xml_prior_distr</a></td>
<td>Represent the site models as XML</td></tr>
<tr><td style="width: 25%;"><a href="site_models_to_xml_state.html">site_models_to_xml_state</a></td>
<td>Converts one or more clock models to the 'state' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="site_models_to_xml_tracelog.html">site_models_to_xml_tracelog</a></td>
<td>Creates the site models' XML for the tracelog section</td></tr>
<tr><td style="width: 25%;"><a href="site_model_to_xml_lh_distr.html">site_model_to_xml_lh_distr</a></td>
<td>Converts a site model to XML, used in the 'siteModel' section</td></tr>
<tr><td style="width: 25%;"><a href="site_model_to_xml_operators.html">site_model_to_xml_operators</a></td>
<td>Converts a site model to XML, used in the 'operators' section</td></tr>
<tr><td style="width: 25%;"><a href="site_model_to_xml_prior_distr.html">site_model_to_xml_prior_distr</a></td>
<td>Converts a site model to XML, used in the 'prior' section</td></tr>
<tr><td style="width: 25%;"><a href="site_model_to_xml_state.html">site_model_to_xml_state</a></td>
<td>Converts a site model to XML, used in the 'state' section</td></tr>
<tr><td style="width: 25%;"><a href="site_model_to_xml_subst_model.html">site_model_to_xml_subst_model</a></td>
<td>Converts a site model to XML, used in the 'substModel' section</td></tr>
<tr><td style="width: 25%;"><a href="site_model_to_xml_tracelog.html">site_model_to_xml_tracelog</a></td>
<td>Creates the site model's XML for the tracelog section</td></tr>
</table>

<h2><a name="T">-- T --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="tree_models_to_xml_tracelog.html">tree_models_to_xml_tracelog</a></td>
<td>Creates the tree models' XML for the tracelog section</td></tr>
<tr><td style="width: 25%;"><a href="tree_priors_to_xml_operators.html">tree_priors_to_xml_operators</a></td>
<td>Creates the XML of a list of one or more tree priors, as used in the 'operators' section</td></tr>
<tr><td style="width: 25%;"><a href="tree_priors_to_xml_prior_distr.html">tree_priors_to_xml_prior_distr</a></td>
<td>Creates the distribution section in the prior section of the distribution section of a BEAST2 XML parameter file. These lines start with '&lt;distribution id='</td></tr>
<tr><td style="width: 25%;"><a href="tree_priors_to_xml_state.html">tree_priors_to_xml_state</a></td>
<td>Converts one or more tree priors to the 'state' section of the XML as text</td></tr>
<tr><td style="width: 25%;"><a href="tree_priors_to_xml_tracelog.html">tree_priors_to_xml_tracelog</a></td>
<td>Creates the tree priors' XML for the tracelog section</td></tr>
<tr><td style="width: 25%;"><a href="tree_prior_to_xml_operators.html">tree_prior_to_xml_operators</a></td>
<td>Creates the XML of a tree prior, as used in the 'operators' section</td></tr>
<tr><td style="width: 25%;"><a href="tree_prior_to_xml_prior_distr.html">tree_prior_to_xml_prior_distr</a></td>
<td>Creates the distribution section in the prior section of the distribution section of a BEAST2 XML parameter file. These lines start with '&lt;distribution id='</td></tr>
<tr><td style="width: 25%;"><a href="tree_prior_to_xml_state.html">tree_prior_to_xml_state</a></td>
<td>Creates the XML of a tree prior, as used in the 'state' section</td></tr>
<tr><td style="width: 25%;"><a href="tree_prior_to_xml_tracelog.html">tree_prior_to_xml_tracelog</a></td>
<td>Creates the tree prior's XML for the tracelog section</td></tr>
</table>

<h2><a name="Y">-- Y --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="yule_tree_prior_to_xml_prior_distr.html">yule_tree_prior_to_xml_prior_distr</a></td>
<td>Creates the 'prior' section in the prior section of the prior section of the distribution section of a BEAST2 XML parameter file for a Yule tree prior</td></tr>
</table>
</body></html>
