#' Creates the map section of a BEAST2 XML parameter file
#' @inheritParams create_beast2_input
#' @param ids the IDs of the alignments (can be extracted from
#'   their FASTA filesnames using \code{\link{get_ids}})
#' @author Richel J.C. Bilderbeek
create_beast2_input_init <- function(
  ids,
  initial_phylogenies
) {
  testit::assert(class(initial_phylogenies) == "multiPhylo" ||
    is.na(initial_phylogenies))
  testit::assert(beautier::are_ids(ids))
  testit::assert(length(ids) == length(initial_phylogenies))

  text <- NULL

  # From https://www.beast2.org/fix-starting-tree/:
  #
  # if there is an initaliser generated by BEAUti of the form
  #
  #   <init estimate="false" id="RandomTree.t:xxx" initial="@Tree.t:xxx"
  #     spec="beast.evolution.tree.RandomTree" taxa="@xxx">
  #   </init>
  #
  # remove this element from the file, otherwise the tree will be Newick tree
  #   will be overwritten by a random tree.
  #
  # In other words: bluntly remove it
  n <- length(ids)
  for (i in seq(1, n)) {
    phylogeny <- initial_phylogenies[[i]]
    id <- ids[i]
    if (!is_phylo(phylogeny)) {
      text <- c(text, "")
      text <- c(
        text,
        beautier::indent(rnd_phylo_to_xml_init(id), n_spaces = 4)
      )
    }
  }
  text
}
